packages/grpc-js/.eslintrc

```diff
@@ -0,0 +1,10 @@
+{
+  "root": true,
+  "extends": "./node_modules/gts",
+  "rules": {
+    "@typescript-eslint/no-explicit-any": "off",
+    "node/no-unpublished-import": ["error", {
+        "tryExtensions": [".ts", ".js", ".json", ".node"]
```

Comment:
We might want a `convertPath` option here. I'm not sure because I'm not entirely clear on why this change makes it work.

Reply:
If I change it to something like
```json
        "convertPath": {
          "src/*.ts": ["^src/(.+?)\\.ts$", "build/src/$1.js"]
        }
```
The rule will ignore every import statement (can be tested by adding `import { mockFunction } from '../test/common';` to any `src/*.ts` file). Adding `".ts"` to the extensions makes it work correctly. Same behavior for `node/no-unpublished-require`.

But a path transform is needed, as `const { version: clientVersion } = require('../../package.json');` raises `error    "../../package.json" is not published                        node/no-unpublished-require`, but `import { version as clientVersion } from '../../package.json';` cannot be compiled.

Reply:
I don't have strong opinions here, as long as you make it pass. I would even be OK with disabling this check if it doesn't work otherwise. I think it would be unlikely to be a problem.

Reply:
Realized, that only the flag resolveJsonModule missed for compiling with `import` statement. As the `package.json` will also be copied in the build folder, the path `../../package.json` was fine, but `../package.json` works also like a charm, including typescript support.